### ISSUE TITLE
AutoComplete drop down closes before reaching the end of options list

### ISSUE THREAD

**Issue by husseinshaib1** [Megabit/Blazorise#4770]
**Describe the bug**
I am implementing autocomplete control with virtualization. When user scrolls down into the drop down more options are loaded from the API. It works fine but when scrolling down at a specific point the drop down closes after retrieving around 50 items or so.

**To Reproduce**

1. add auto complete control to your page
2. set the "Virtualize" option 
3. get data from api based on user demand
4. notice the drop closes suddenly after some scrolls(quick ones)

**Expected behavior**
The drop down should display loaded data from API and remain opened so user can select an option he is searching for.

**Comment by David-Moreira** [comment]
@husseinshaib1 
The reason why we specifically ask for repros, is because there sometimes can be subtle differences that can make the control act differently that just a few generalized steps might not always be enough to reproduce. 
And then we might loose precious time trying to reproduce behavior unsuccessfully.

Can you at least try to reproduce the issue here and tells us the steps based on this demo?
https://bootstrapdemo.blazorise.com/tests/autocomplete

I tried unsuccessfully but I might be doing something wrong.

**Comment by husseinshaib1** [comment]
I have updated the bug with the repro

**Comment by David-Moreira** [comment]
@husseinshaib1 `Virtualize` is getting confused because you are not giving it accurate information.
The `TotalItems` available are 72, the way you filter data is by just skipping the offset, you don't need to do any calculation.
Please correct the following and try again. Also if the purpose if to just use InMemory data you don't need to define `ReadData`.

`TotalItems="72"`
```
    private void OnHandleReadData( AutocompleteReadDataEventArgs autocompleteReadDataEventArgs )
    {

        if ( !autocompleteReadDataEventArgs.CancellationToken.IsCancellationRequested )
        {

            if ( !autocompleteReadDataEventArgs.CancellationToken.IsCancellationRequested )
            {
                Temp = DataPoints.Skip( autocompleteReadDataEventArgs.VirtualizeOffset ).Take( autocompleteReadDataEventArgs.VirtualizeCount ).ToList();
            }
        }
    }
```

**Comment by husseinshaib1** [comment]
yes I got it. what I was doing is the normal pagination where you will always get page size and page index multiply them in order to get the number of pages I should skip in my retrieval from database.
But the virtualize feature already handles this and gives us the offset that we should use.
Also, I tried to create a repro just for you to explain what is happening it is not the way I am using it.

Thank you guys for the awesome support!
you are the best
